lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,353 @@
+""" API implementation for course-oriented interactions. """
+
+import logging
+
+from django.http import Http404
+from rest_framework.exceptions import ParseError
+from rest_framework.generics import RetrieveAPIView, ListAPIView
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.lib.api.views import PaginatedListAPIViewWithKeyHeaderPermissions, ApiKeyHeaderPermissionMixin
+
+from course_api.courseware_access import get_course, get_course_child, get_course_descriptor
+from course_api.v0.serializers import CourseSerializer, GradedContentSerializer, GradingPolicySerializer, \
+    CourseContentSerializer
+
+
+log = logging.getLogger(__name__)
+
+
+class CourseContentMixin(object):
+    default_depth = 0
+    serializer_class = CourseContentSerializer
+    lookup_field = 'course_id'
+
+    def get_serializer_context(self):
+        context = super(CourseContentMixin, self).get_serializer_context()
+        context['course_id'] = self.kwargs.get('course_id')
+        return context
+
+    def depth(self, request):
+        try:
+            return int(request.QUERY_PARAMS.get('depth', self.default_depth))
+        except ValueError:
+            raise ParseError
+
+    def get_course_or_404(self, request, course_id):
+        depth = self.depth(request)
+        course_descriptor = get_course(course_id, depth=depth)
+
+        if not course_descriptor:
+            raise Http404
+
+        return course_descriptor
+
+    def fetch_children(self, content_descriptor, depth=0):
```

Comment:
@gwprice After a day and a half dealing with devstack, I finally switched all views to using serializers. This method is needed to retrieve the children to a specified depth; otherwise, the serializer would traverse the entire content tree.